web-ui/src/main/resources/catalog/components/metadataactions/RelatedDirective.js

```diff
@@ -384,6 +389,9 @@
                 return;
               }
 
+              // Todo 
+              // Add add check for gnCurrentEdit.metadata.draft === "y" and append "approved=false" to url
+              
```

Comment:
I've tested the following code below https://github.com/geonetwork/core-geonetwork/pull/7248/files#diff-56dcf6dc1a8abde25966a791bff4c603791e3a402d723b17a042c8e444daecffL405, probably needs some improvements on the protocol and relations to check, but for online resources of uploaded files, looks working fine:

```
              // For draft version append "approved=false" to url
              if ((scope.md.draft === 'y') && (idx === 'onlines')) {
                for (var i = 0; i < scope.relations[idx].length; i++) {
                  if (scope.relations[idx][i].protocol === 'WWW:DOWNLOAD') {
                    scope.relations[idx][i].url += "?approved=false";
                  }
                }
              }
```

@ianwallen, can you try it and update the pull request if looks ok to you?

Reply:
This fixed the resource links however the thumbnails are still not working.

I tried to change it to the following.

if (scope.md.draft === "y" && (idx === "onlines" || idx === "thumbnails")) {

But that did not fix the issue - So I'm guessing that thumbnails are somewheres else.